src/ucs/sys/sock.c

```diff
@@ -537,9 +537,13 @@ ucs_status_t ucs_sockaddr_get_ifname(int fd, char *ifname_str, size_t max_strlen
         return UCS_ERR_IO_ERROR;
     }
 
+
     for (ifa = ifaddrs; ifa != NULL; ifa = ifa->ifa_next) {
         sa = (struct sockaddr*) ifa->ifa_addr;
-        if (((sa->sa_family == AF_INET) ||(sa->sa_family == AF_INET6)) && 
+
+        if(sa == NULL)
+            ucs_debug("NULL ifaddr encountered with ifa_name: %s", ifa->ifa_name);
+        if (sa != NULL && ((sa->sa_family == AF_INET) ||(sa->sa_family == AF_INET6)) && 
```

Comment:
code style: pls add ( ) around `sa != NULL` check

Reply:
btw this could be `else if if (((sa->sa_family == ...` instead of checking again if sa!=NULL